src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -214,10 +226,21 @@ public void VerifyTypeIL(string typeName, string expected)
 
             string mainModuleName = Emit(testEnvironment, manifestResources, emitOptions);
             _allModuleData = testEnvironment.GetAllModuleData();
-            testEnvironment.Verify(peVerify);
+
+            try
+            {
+                testEnvironment.Verify(peVerify);
 #if NETCOREAPP
-            ILVerify(peVerify);
+                ILVerify(peVerify);
 #endif
+            }
+            catch (Exception) when (peVerify is Verification.PassesOrFailFast)
+            {
+                var il = DumpIL();
+                Console.WriteLine(il);
+
+                Environment.FailFast("Investigating flaky IL verification issue. Tracked by https://github.com/dotnet/roslyn/issues/63782");
```

Comment:
Why use `Environment.FailFast` instead of `Assert.True(false, "Message")`? That will show up nicer in test logs.

Reply:
This will generate a crash dump (which the test infrastructure should capture in artifacts) in addition to dumping IL.